encoders/build.gradle

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2017 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+apply plugin: 'scala'
+
+compileScala.options.encoding = 'UTF-8'
+// fix scala+java mix to all use compileScala which uses correct dependency order
+sourceSets.main.scala.srcDir 'src/main/java'
+sourceSets.test.scala.srcDirs = ['src/test/java', 'src/test/scala',
+                                 'src/dunit/java', 'src/dunit/scala']
+sourceSets.main.java.srcDirs = []
+sourceSets.test.java.srcDirs = []
+
+dependencies {
+
+  compileOnly 'org.scala-lang:scala-library:' + scalaVersion
+  compileOnly 'org.scala-lang:scala-reflect:' + scalaVersion
+
+  compileOnly("org.apache.spark:spark-hive_${scalaBinaryVersion}:${sparkVersion}")
```

Comment:
Where is this used and why is this compileOnly?

Reply:
Looking at the code I think proper dependencies on spark-core, spark-catalyst and spark-sql should be added. These can be normal compile depdendencies with no shadowJar.

Reply:
- spark-hive dependency is needed for the classes like InternalRow, DataTypes IntegerType, StringType etc classes. 
- Removed the compileOnly 'org.scala-lang:scala-reflect:' + scalaVersion
- Added the spark-core, spark-catalyst, and spark-sql dependecies.